esp-hal/src/clock/mod.rs

```diff
@@ -591,65 +599,136 @@ impl Clocks {
     }
 }
 
-/// Control the radio peripheral clocks
 #[cfg(any(bt, ieee802154, wifi))]
+/// Tracks which modems currently request an active PHY clock.
+static PHY_CLOCK_REF_COUNTERS: critical_section::Mutex<Cell<u8>> =
+    critical_section::Mutex::new(Cell::new(0));
+
+// These functions are moved out of the trait to prevent monomorphization for
+// every modem clock controller. If that doens't really make sense, this can be
+// moved back.
```

Comment:
This is fine as a fact, although I'm not sure how much value this comment has in the source :)